### PR TITLE
Toolbar enhancements

### PR DESCRIPTION
**Pull request by francescospissu**
### Motivation
The toolbar needs to be updated with new icons, different sizes and other graphical improvements to reflect the new design system.

### Change description
Graphic updates based on new specifications.

### Reviewer checklist

* [ ] PR addresses a single concern.
* [ ] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-ide/pulls) before creating one)
* [ ] PR title and description are properly filled.
* [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: arduino-ide-extension/src/browser/style/main.css [hunk 1]
```diff
@@ -3,7 +3,8 @@
 }
 
 .p-TabBar-toolbar .item.arduino-tool-item {
-    margin-left: 3px;
+    margin-right: 8px;
+    margin-left: unset;
```
Comment (AlbyIanna) [inline arduino-ide-extension/src/browser/style/main.css:7]:
```suggestion
    margin: unset;
```

File: arduino-ide-extension/src/browser/style/main.css [hunk 2]
```diff
@@ -106,11 +102,20 @@
 }
 
 .p-TabBar-toolbar.theia-arduino-toolbar {
+    align-items: center;
     z-index: 0;
```
Comment (AlbyIanna) [inline arduino-ide-extension/src/browser/style/main.css:106]:
```suggestion
    align-items: center;
    gap: 8px;
    margin: 0 12px;
    padding: 0;
    z-index: 0;
```

File: arduino-ide-extension/src/browser/style/main.css [hunk 3]
```diff
@@ -106,11 +102,20 @@
...
 }
 
+.p-TabBar-toolbar {
+    margin-right: 12px;
+}
```
Comment (AlbyIanna) [inline arduino-ide-extension/src/browser/style/main.css:111]:
```suggestion
```

File: arduino-ide-extension/src/browser/style/main.css [hunk 4]
```diff
@@ -106,11 +102,20 @@
...
+
+:root {
+    --theia-private-menubar-height: 40px; /* set the topbar height */
+}
+
 #theia-top-panel .p-TabBar-toolbar {
```
Comment (AlbyIanna) [inline arduino-ide-extension/src/browser/style/main.css:117]: remove this rule

File: arduino-ide-extension/src/browser/style/main.css [hunk 5]
```diff
@@ -106,11 +102,20 @@
...
-    padding-left: 4px !important; /* moves the `verify`, upload and other toolbar items to the left */
+    padding-left: 12px !important; /* moves the `verify`, upload and other toolbar items to the left */
 }
 
 .p-Widget .p-MenuBar {
```
Comment (AlbyIanna) [inline arduino-ide-extension/src/browser/style/main.css:121]: remove this rule

File: arduino-ide-extension/src/browser/style/main.css [hunk 6]
```diff
@@ -120,6 +125,7 @@
 #theia-top-panel .p-TabBar-toolbar.theia-arduino-toolbar.right {
     justify-content: flex-start;
     min-width: 100px;
+    margin-right: 4px;
```
Comment (AlbyIanna) [inline arduino-ide-extension/src/browser/style/main.css:128]:
```suggestion
```

File: arduino-ide-extension/src/browser/style/main.css [hunk 7]
```diff
@@ -128,7 +134,8 @@
 }
 
 .arduino-toolbar-tooltip {
-    margin-left: 10px;
+    margin-left: 10px;    
+    margin-right: 4px;
```
Comment (AlbyIanna) [inline arduino-ide-extension/src/browser/style/main.css:138]:
```suggestion
```

**Review by AlbyIanna — CHANGES_REQUESTED** [review]
Overall it's good, nice job!
I've just made some suggestions to have a cleaner code (in particular the CSS)

File: arduino-ide-extension/src/browser/style/main.css (hunk 1 above)
Comment (fstasi) [reply arduino-ide-extension/src/browser/style/main.css:7]: be careful here. @AlbyIanna unsetting the margin is different from unsettling the margin-left.

If the `margin-left` is assigned / inherited, `margin: unset;` will not clear it

File: arduino-ide-extension/src/browser/style/main.css (hunk 4 above)
Comment (fstasi) [reply arduino-ide-extension/src/browser/style/main.css:117]: why?

File: arduino-ide-extension/src/browser/style/main.css (hunk 1 above)
Comment (AlbyIanna) [reply arduino-ide-extension/src/browser/style/main.css:7]: You're right,  it's probably more correct to do like this:
```suggestion
    margin-left: 0;
```

**Review by fstasi — APPROVED** [review]

File: arduino-ide-extension/src/browser/style/main.css (hunk 7 above)
Comment (francescospissu) [reply arduino-ide-extension/src/browser/style/main.css]: Removed [here](https://github.com/arduino/arduino-ide/pull/1194/commits/0c84f8b39beee98ec2c6ed89482250b8a80ac652).

File: arduino-ide-extension/src/browser/style/main.css (hunk 6 above)
Comment (francescospissu) [reply arduino-ide-extension/src/browser/style/main.css]: Removed [here](https://github.com/arduino/arduino-ide/pull/1194/commits/0c84f8b39beee98ec2c6ed89482250b8a80ac652).

File: arduino-ide-extension/src/browser/style/main.css (hunk 3 above)
Comment (francescospissu) [reply arduino-ide-extension/src/browser/style/main.css]: Removed [here](https://github.com/arduino/arduino-ide/pull/1194/commits/0c84f8b39beee98ec2c6ed89482250b8a80ac652).

**Review by AlbyIanna — APPROVED** [review]
Thanks for the changes!

Look at this wonderful new UI 🤩 

LGTM 🚀

**Review by 91volt — APPROVED** [review]

**Pull request merged**